packages/react-i18n/src/i18n.ts

```diff
@@ -364,6 +368,23 @@ export class I18n {
     return Boolean(easternNameOrderFormatter);
   }
 
+  private formatCurrencyAuto(
+    amount: number,
+    options: Intl.NumberFormatOptions = {},
+  ): string {
+    // use the short format if we can't determine a currency match, or if the
+    // currencies match, use explicit when the currencies definitively do not
+    // match.
+    const formatShort =
+      !options.currency ||
```

Comment:
I usually prefer to be explicit here and check if `options.currency == null`. 

If `currency` is an empty string, the condition will be true and I'm not sure if it should 🤔

Reply:
yea good point